core/action/action.go

```diff
@@ -41,6 +43,19 @@ func NewAction(actionName, binary, scriptContent string, delayInMs int) (*Action
 	return scriptInfo, nil
 }
 
+func NewRemoteAction(actionName, host string, delayInMs int) (*Action, error) {
+
+	if strings.TrimSpace(actionName) == "" {
+		return nil, errors.New("empty action name passed")
+	}
+
+	if !hasHttpPrefix(host) {
```

Comment:
Can we use the following to validate url pls? 
```
func validateURL(u string) bool {
    _, err := url.ParseRequestURI(u)
    return err == nil
}
```

Reply:
incorporated